android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java

```diff
@@ -553,7 +575,7 @@ private void setAudioFocus(boolean focus) {
         public void onReceive(Context context, Intent intent) {
 //            audioManager.setSpeakerphoneOn(true);
             if (Intent.ACTION_HEADSET_PLUG.equals(intent.getAction())) {
-                audioManager.setSpeakerphoneOn(!audioManager.isWiredHeadsetOn());
+               audioManager.setSpeakerphoneOn(false);
```

Comment:
I believe this is fired both when plugged and unplugged; it might be worth pulling the code you added in the block above into its own function and just re-calling it here.